inference-engine/src/gna_plugin/layers/gna_layer_type.cpp

```diff
@@ -60,6 +60,22 @@ bool GNAPluginNS::AreLayersSupported(InferenceEngine::CNNNetwork& network, std::
                                                                 ", and batch size(" + std::to_string(batch_size) + ") != 1 not supported";
                                                    check_result =  false;
                                                }
+                                               if (LayerInfo(layer).isFullyConnected()) {
+                                                   size_t output_batch_size = LayerInfo(layer).getBatchSize();
```

Comment:
It a quite a bit strange to check output size instead of input sizes.

Reply:
Checking the output in this case is easier than the input. I see no reason not to use output validation.